### PR TITLE
Media Browser Rehaul (Refs SHOOP-568, SHOOP-503, SHOOP-1113)

### PR DESCRIPTION
**Pull request by akx**

### CODE REVIEW

**Comment by akx** [comment]
As usual, there's a few small bugfixes at the top of the PR.

File: shoop/admin/modules/media/static_src/media/browser/FileUpload.js [hunk 1]
```diff
@@ -6,69 +6,145 @@
...
+    }
+    setTimeout(processQueue, 50);  // Continue soon.
+    var messageText = null;
+    try {
+        const responseJson = JSON.parse(xhr.responseText);
+        if (responseJson && responseJson.message) {
+            messageText = responseJson.message;
+        }
+    } catch (e) {
+        console.log(e); // invalid JSON? pffff.
```
Comment (Pikkupomo) [inline shoop/admin/modules/media/static_src/media/browser/FileUpload.js]: Show some error from this also? for user that is

File: shoop/admin/modules/media/static_src/media/browser/FileUpload.js [hunk 2]
```diff
@@ -6,69 +6,145 @@
...
+    file.progress = 0;
+    file.status = "uploading";
+
+    const formData = new FormData();
+    formData.append("file", file.nativeFile);
+    const xhr = new XMLHttpRequest();
+    xhr.open("POST", file.url);
+    xhr.setRequestHeader("X-CSRFToken", window.ShoopAdminConfig.csrf);
+    xhr.onreadystatechange = function() {
+        if (xhr.readyState !== 4) {
```
Comment (Pikkupomo) [inline shoop/admin/modules/media/static_src/media/browser/FileUpload.js]: maybe add comment what this magical `4` is for future readers?
`// 4: request finished and response is ready`

File: shoop/admin/modules/media/static_src/media/browser/FileUpload.js (hunk 1 above)
Comment (akx) [inline shoop/admin/modules/media/static_src/media/browser/FileUpload.js]: It's only useful for developers, hence the console.log

**Comment by akx** [comment]
@Pikkupomo: fyi, rebased & removed that cssnano change.

File: shoop/admin/modules/media/static_src/media/browser/FileUpload.js (hunk 2 above)
Comment (akx) [inline shoop/admin/modules/media/static_src/media/browser/FileUpload.js]: Fixup added.

File: shoop/admin/modules/media/static_src/media/browser/components/wrapFileLink.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * LICENSE file in the root directory of this source tree.
+ */
+const m = require("mithril");
+const getPickId = require("../util/getPickId");
+
+export default function(file, tag = "a", content = file.name) {
+    const attrs = {href: file.url, target: "_blank"};
+    const pickId = getPickId();
+    if (pickId) {
+        console.log(pickId);
```
Comment (Pikkupomo) [inline shoop/admin/modules/media/static_src/media/browser/components/wrapFileLink.js]: stray log?

Comment (akx) [inline shoop/admin/modules/media/static_src/media/browser/components/wrapFileLink.js]: Yerp. Fixup added.

File: shoop/admin/modules/media/static_src/media/browser/FileUpload.js (hunk 1 above)
Comment (Pikkupomo) [inline shoop/admin/modules/media/static_src/media/browser/FileUpload.js]: ok

**Comment by akx** [comment]
Rebased (to fold in fixups). No other changes, @Pikkupomo

**Comment by akx** [comment]
Rebased again because raisins.

**Pull request merged**